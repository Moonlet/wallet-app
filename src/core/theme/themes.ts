export enum Theme {
    dark = 'dark',
    light = 'light'
}
