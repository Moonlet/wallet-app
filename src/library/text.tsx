import React from 'react';
import ReactNative from 'react-native';

const defaultStyle = {
    color: '#FFFFFF',
    fontFamily: 'System',
    fontSize: 16
};

const mergeStyle = (baseStyle: any, newStyle: any) => {
    let style;

    if (Array.isArray(newStyle)) {
        style = newStyle.slice();
        style.unshift(baseStyle);
    } else {
        style = { ...baseStyle, ...newStyle };
    }
    return style;
};

const getProps = (props: any) => ({
    children: props.children
});

const Text = (props: any) => {
    return <ReactNative.Text style={mergeStyle(defaultStyle, props.style)} {...getProps(props)} />;
};

const TextSmall = (props: any) => {
    delete props.style;

    return (
        <Text
            {...getProps(props)}
            style={mergeStyle(
                {
                    fontSize: 12
                },
                props.style
            )}
        />
    );
};

export { Text, TextSmall };
