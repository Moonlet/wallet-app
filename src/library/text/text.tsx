import React from 'react';
import * as ReactNative from 'react-native';

import { formatNumber, INumberFormatOptions } from '../../core/utils/format-number';
import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';

export interface ITextProps {
    children?: any;
    style?: any;
    darker?: boolean;
    small?: boolean;
    format?: INumberFormatOptions;
    styles: ReturnType<typeof stylesProvider>;
    numberOfLines?: number;
    ellipsizeMode?: 'head' | 'middle' | 'tail' | 'clip';
}

const styleModifiers = ['darker', 'small', 'large'];

const getStyle = (props: ITextProps) => {
    // default text style
    let style = [props.styles.default];

    // props texts modifiers
    styleModifiers.forEach((modifier: string) => {
        // @ts-ignore
        props[modifier] && props.styles[modifier] && style.push(props.styles[modifier]);
    });

    // style from props
    if (props.style) {
        if (Array.isArray(props.style)) {
            style = style.concat(props.style);
        } else {
            style.push(props.style);
        }
    }

    return style;
};

export const TextComponent = (props: ITextProps) => {
    let text = props.children;

    if (props.format) {
        const amount = parseFloat(text);
        text = formatNumber(amount, props.format);
    }
    return (
        <ReactNative.Text
            style={getStyle(props)}
            numberOfLines={props.numberOfLines}
            ellipsizeMode={props.ellipsizeMode}
        >
            {text}
        </ReactNative.Text>
    );
};

export const Text = withTheme(stylesProvider)(TextComponent);
