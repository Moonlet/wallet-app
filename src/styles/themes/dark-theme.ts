import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
