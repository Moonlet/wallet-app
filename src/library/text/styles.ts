import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { normalize } from './text';

export default (theme: ITheme) =>
    StyleSheet.create({
        default: {
            color: theme.colors.text,
            fontFamily: 'System',
            fontSize: normalize(theme.fontSize.regular)
        },
        darker: {
            color: theme.colors.textSecondary
        },
        small: {
            fontSize: normalize(theme.fontSize.small)
        },
        large: {
            fontSize: normalize(theme.fontSize.large)
        }
    });
