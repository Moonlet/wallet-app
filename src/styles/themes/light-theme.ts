import { ITheme } from '../../core/theme/itheme';
import { darkTheme } from './dark-theme';

export const lightTheme: ITheme = darkTheme;
