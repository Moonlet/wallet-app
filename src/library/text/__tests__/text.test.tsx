import React from 'react';
import { TextComponent, ITextProps } from '../text';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';

const props: ITextProps = {
    styles: styleProvider(darkTheme)
};

describe('text library components', () => {
    it('renders correctly regular text', () => {
        const wrapper = shallow(<TextComponent {...props}>dummy</TextComponent>);
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('renders correctly formatted text', () => {
        const wrapper = shallow(
            <TextComponent {...props} format={{ maximumFractionDigits: 3 }}>
                100.133333333
            </TextComponent>
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('renders correctly small text', () => {
        const wrapper = shallow(
            <TextComponent {...props} small>
                test
            </TextComponent>
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('renders correctly text with style provided via prop', () => {
        const wrapper = shallow(
            <TextComponent {...props} style={{ fontSize: 34 }}>
                test
            </TextComponent>
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('renders correctly text with multiple styles', () => {
        const wrapper = shallow(
            <TextComponent {...props} style={[{ fontSize: 34 }, { color: 'red' }]}>
                test
            </TextComponent>
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
