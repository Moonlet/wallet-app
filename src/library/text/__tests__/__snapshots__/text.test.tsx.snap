// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text library components renders correctly formatted text 1`] = `
"<Text style={{...}}>
  100.133
</Text>"
`;

exports[`text library components renders correctly regular text 1`] = `
"<Text style={{...}}>
  dummy
</Text>"
`;

exports[`text library components renders correctly small text 1`] = `
"<Text style={{...}}>
  test
</Text>"
`;

exports[`text library components renders correctly text with multiple styles 1`] = `
"<Text style={{...}}>
  test
</Text>"
`;

exports[`text library components renders correctly text with style provided via prop 1`] = `
"<Text style={{...}}>
  test
</Text>"
`;
